# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: toolkit/components/thumbnails/PageThumbsWorker.js
# Commit: a400e87a7082
# Full Hash: a400e87a708271cc1830228b4d05ff19b455136c
# Author: Emma Malysz <emalysz@mozilla.com>
# Date: 2021-05-22 03:53:20
# Description:
#   Bug 1711634, reduce number of AsyncShutdownTimeout crashes from PageThumbs r=barret
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D115407
# ==============================================================================

diff -r 978c022b695d -r a400e87a7082 toolkit/components/thumbnails/PageThumbsWorker.js
--- a/toolkit/components/thumbnails/PageThumbsWorker.js	Fri May 21 23:21:58 2021 +0300
+++ b/toolkit/components/thumbnails/PageThumbsWorker.js	Fri May 21 20:27:16 2021 +0000
@@ -134,18 +134,16 @@
 
   async wipe(path) {
     let children = await IOUtils.getChildren(path);
-    for (let entry of children) {
-      try {
-        await IOUtils.remove(entry);
-      } catch (ex) {
-        // If a file cannot be removed, we should still continue.
-        // This can happen at least for any of the following reasons:
-        // - access denied;
-        // - file has been removed recently during a previous wipe
-        //  and the file system has not flushed that yet (yes, this
-        //  can happen under Windows);
-        // - file has been removed by the user or another process.
-      }
+    try {
+      await Promise.all(children.map(entry => IOUtils.remove(entry)));
+    } catch (ex) {
+      // If a file cannot be removed, we should still continue.
+      // This can happen at least for any of the following reasons:
+      // - access denied;
+      // - file has been removed recently during a previous wipe
+      //  and the file system has not flushed that yet (yes, this
+      //  can happen under Windows);
+      // - file has been removed by the user or another process.
     }
   },
 
